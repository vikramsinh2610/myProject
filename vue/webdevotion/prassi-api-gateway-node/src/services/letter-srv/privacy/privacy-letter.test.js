const PrivacyLetter = require('./privacy-letter');

describe('PrivacyLetter', () => {
  test('should have these properties', () => {
    const letter = new PrivacyLetter({
      _id: '01-2020',
      status: 'wip',
      promoterId: 'f34g3df-f3c3x2ed-d232r4g5h5-d21s',
      promoterSerialNumber: 'AZ123456',
      promoterDisplayName: 'Pippo De Peppis',
      description: 'Bla bla bla',
      signatureDate: new Date(),
      attachmentIds: ['a', 'b'],
      agreements: [],
    });
    expect(letter).toHaveProperty('_id');
    expect(letter).toHaveProperty('status');
    expect(letter).toHaveProperty('promoterId');
    expect(letter).toHaveProperty('promoterSerialNumber');
    expect(letter).toHaveProperty('promoterDisplayName');
    expect(letter).toHaveProperty('type');
    expect(letter).toHaveProperty('description');
    expect(letter).toHaveProperty('signatureDate');
    expect(letter).toHaveProperty('attachmentIds');
    expect(letter).toHaveProperty('agreements');
  });

  test('should return it s JSON Schema', () => {
    expect(PrivacyLetter.getJSONSchema()).toMatchSnapshot();
  });
});
