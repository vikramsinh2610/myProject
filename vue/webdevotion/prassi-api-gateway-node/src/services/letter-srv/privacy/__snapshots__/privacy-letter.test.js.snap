// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivacyLetter should return it s JSON Schema 1`] = `
Object {
  "description": "Bonus Letter",
  "properties": Object {
    "_id": Object {
      "type": "string",
    },
    "agreements": Object {
      "items": Object {
        "properties": Object {
          "agree": Object {
            "type": "boolean",
          },
          "content": Object {
            "type": "string",
          },
          "name": Object {
            "type": "string",
          },
        },
        "required": Array [
          "name",
          "content",
          "agree",
        ],
        "type": "object",
      },
      "type": "array",
    },
    "attachmentIds": Object {
      "description": "Attachements ID",
      "items": Object {
        "type": "string",
      },
      "type": "array",
    },
    "description": Object {
      "type": "string",
    },
    "didActiveDate": Object {
      "pattern": "^((19|20)[0-9]{2})-(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])T(2[0-3]|[01][0-9]):([0-5][0-9]):([0-5][0-9])([.])([0-9]{3})Z",
      "type": "string",
    },
    "didCreateDate": Object {
      "pattern": "^((19|20)[0-9]{2})-(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])T(2[0-3]|[01][0-9]):([0-5][0-9]):([0-5][0-9])([.])([0-9]{3})Z",
      "type": "string",
    },
    "didDeleteDate": Object {
      "pattern": "^((19|20)[0-9]{2})-(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])T(2[0-3]|[01][0-9]):([0-5][0-9]):([0-5][0-9])([.])([0-9]{3})Z",
      "type": "string",
    },
    "didExpireDate": Object {
      "pattern": "^((19|20)[0-9]{2})-(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])T(2[0-3]|[01][0-9]):([0-5][0-9]):([0-5][0-9])([.])([0-9]{3})Z",
      "type": "string",
    },
    "promoterDisplayName": Object {
      "description": "Promoter full name",
      "type": "string",
    },
    "promoterId": Object {
      "description": "Promoter ID",
      "type": "string",
    },
    "promoterSerialNumber": Object {
      "description": "Promoter Serial Number",
      "type": "string",
    },
    "signatureDate": Object {
      "pattern": "^((19|20)[0-9]{2})-(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])T(2[0-3]|[01][0-9]):([0-5][0-9]):([0-5][0-9])([.])([0-9]{3})Z",
      "type": "string",
    },
    "status": Object {
      "description": "Status of the letter",
      "enum": Array [
        "wip",
        "active",
        "inactive",
        "expired",
        "deleted",
      ],
      "type": "string",
    },
    "type": Object {
      "enum": Array [
        "none",
        "job",
        "commissioning-pa",
        "commissioning-pas",
        "bonus-pa",
        "bonus-pas",
        "rappel-pa",
        "rappel-pa-2021",
        "rappel-pas",
        "rappel",
        "rappel-2",
        "rappel-3",
        "rappel-5",
        "welcome-bonus",
        "guaranteed",
        "guaranteed-variable",
        "guaranteed-variable-minimum",
        "guaranteed-with-bonus",
        "guaranteed-with-bonus-prepaid",
        "management-fee",
        "privacy",
      ],
      "type": "string",
    },
    "willExpireDate": Object {
      "pattern": "^((19|20)[0-9]{2})-(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])T(2[0-3]|[01][0-9]):([0-5][0-9]):([0-5][0-9])([.])([0-9]{3})Z",
      "type": "string",
    },
  },
  "required": Array [
    "_id",
    "status",
    "promoterId",
    "promoterSerialNumber",
    "promoterDisplayName",
    "type",
    "description",
    "signatureDate",
    "attachmentIds",
    "agreements",
  ],
  "type": "object",
}
`;
